src/Service.GraphQLBuilder/Sql/SchemaConverter.cs

```diff
@@ -31,184 +31,387 @@ public static class SchemaConverter
+                    // relationship. To determine this, compare with the entity's database object.
+                    if (pair.ReferencingDbTable.Equals(databaseObject))
+                    {
+                        isNullableRelationship = sourceDefinition.IsAnyColumnNullable(foreignKeyInfo.ReferencingColumns);
+                        foreignKeyFieldsInEntity.UnionWith(foreignKeyInfo.ReferencingColumns);
+                    }
+                    else
+                    {
+                        isNullableRelationship = sourceDefinition.IsAnyColumnNullable(foreignKeyInfo.ReferencedColumns);
+                    }
+                }
+                else
+                {
+                    throw new DataApiBuilderException(
+                        message: $"No relationship exists between {entityName} and {targetEntityName}",
+                        statusCode: HttpStatusCode.InternalServerError,
+                        subStatusCode: DataApiBuilderException.SubStatusCodes.GraphQLMapping);
+                }
+            }
+
+            INullableTypeNode targetField = relationship.Cardinality switch
+            {
+                Cardinality.One =>
+                    new NamedTypeNode(GetDefinedSingularName(targetEntityName, referencedEntity)),
+                Cardinality.Many =>
+                    new NamedTypeNode(QueryBuilder.GeneratePaginationTypeName(GetDefinedSingularName(targetEntityName, referencedEntity))),
+                _ =>
+                    throw new DataApiBuilderException(
+                        message: "Specified cardinality isn't supported",
+                        statusCode: HttpStatusCode.InternalServerError,
+                        subStatusCode: DataApiBuilderException.SubStatusCodes.GraphQLMapping),
+            };
+
+            if (isNestedMutationSupported && relationship.LinkingObject is not null && entitiesWithManyToManyRelationships is not null)
+            {
+                entitiesWithManyToManyRelationships.Add(new(entityName, targetEntityName));
+            }
+
+            FieldDefinitionNode relationshipField = new(
+                location: null,
+                new NameNode(relationshipName),
+                description: null,
+                new List<InputValueDefinitionNode>(),
+                isNullableRelationship ? targetField : new NonNullTypeNode(targetField),
+                new List<DirectiveNode> {
+                            new(RelationshipDirectiveType.DirectiveName,
+                                new ArgumentNode("target", GetDefinedSingularName(targetEntityName, referencedEntity)),
+                                new ArgumentNode("cardinality", relationship.Cardinality.ToString()))
+                });
+
+            return relationshipField;
+        }
+
+        /// <summary>
+        /// Helper method to generate the list of directives for an entity's object type definition.
```

Comment:
helpful to note this generates and returns `authorize` and `model` directive objects.